focal_loss.py

```diff
@@ -51,6 +53,13 @@ def forward(self, x: Tensor, y: Tensor) -> Tensor:
 
         # compute weighted cross entropy term: -alpha * log(pt)
         log_p = F.log_softmax(x, dim=-1)
+        if self.nll_loss.ignore_index == x.shape[1]:
```

Comment:
"if the ignore index is one more than the index values produced by the mode ..."